datafusion/optimizer/src/utils.rs

```diff
@@ -26,32 +26,38 @@ use datafusion_expr::expr_rewriter::replace_col;
 use datafusion_expr::utils as expr_utils;
 use datafusion_expr::{logical_plan::LogicalPlan, Expr, Operator};
 
+use datafusion_common::tree_node::Transformed;
 use log::{debug, trace};
 
 /// Convenience rule for writing optimizers: recursively invoke
 /// optimize on plan's children and then return a node of the same
 /// type. Useful for optimizer rules which want to leave the type
 /// of plan unchanged but still apply to the children.
 /// This also handles the case when the `plan` is a [`LogicalPlan::Explain`].
-///
-/// Returning `Ok(None)` indicates that the plan can't be optimized by the `optimizer`.
 pub fn optimize_children(
```

Comment:
Rather than changing this API, I recommend we deprecate it too (and tell people to use `OptimizerRule::apply_order with ApplyOrder::BottomUp` instead)? 

That way we will avoid API churn

Reply:
Done.